### PR TITLE
updated the sample to use share API instead of _prepare_to_copy

### PR DESCRIPTION
**Pull request by fkriti**
# Description

# Checklist

- [ ] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/python/assets/assets-in-registry/share-data-using-registry.ipynb [hunk 1]
```diff
@@ -440,27 +441,16 @@
...
-    "print(\"data from workspace:\\n\\n\", data_in_workspace)\n",
-    "\n",
-    "# change the format such that the registry understands the data (when you print the data_ready_to_copy object, notice the asset id\n",
-    "data_ready_to_copy = ml_client_workspace.data._prepare_to_copy(\n",
-    "    data_in_workspace, name=\"transformed-nyc-taxt-data-shared-from-ws\", version=version\n",
-    ")\n",
-    "print(\"\\n\\ndata ready to copy:\\n\\n\", data_ready_to_copy)\n",
-    "# copy the data from registry to workspace\n",
-    "ml_client_registry.data.create_or_update(data_ready_to_copy).wait()"
+    "# Sharing data from registry to workspace\n",
```
Comment (vmpham1012) [inline sdk/python/assets/assets-in-registry/share-data-using-registry.ipynb:444]: Should this be: share data from workspace to registry?

Comment (fkriti) [reply sdk/python/assets/assets-in-registry/share-data-using-registry.ipynb]: Fixed

**Review by vmpham1012 — APPROVED** [review]

File: sdk/python/assets/assets-in-registry/share-data-using-registry.ipynb [hunk 2]
```diff
@@ -440,27 +441,22 @@
...
    ]
   }
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python 3.10 - SDK V2",
+   "display_name": "updated",
    "language": "python",
-   "name": "python310-sdkv2"
+   "name": "python3"
```
Comment (singankit) [inline sdk/python/assets/assets-in-registry/share-data-using-registry.ipynb:459]: Shall we keep the metadata like before ?

File: sdk/python/assets/assets-in-registry/share-data-using-registry.ipynb [hunk 3]
```diff
@@ -472,7 +468,12 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.8.8"
+   "version": "3.11.2"
```
Comment (singankit) [inline sdk/python/assets/assets-in-registry/share-data-using-registry.ipynb:471]: Same here

File: sdk/python/assets/assets-in-registry/share-data-using-registry.ipynb [hunk 4]
```diff
@@ -473,6 +469,11 @@
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
    "version": "3.8.8"
+  },
+  "vscode": {
```
Comment (needuv) [inline sdk/python/assets/assets-in-registry/share-data-using-registry.ipynb:473]: remove this as well

**Pull request merged**